pwnlib/rop/rop.py

```diff
@@ -1206,6 +1206,92 @@ def search(self, move = 0, regs = None, order = 'size'):
 
         return result
 
+    def ret2csu(self, edi, rsi, rdx, rbx=0, rbp=0, r12=0, r13=0, r14=0, r15=0, call=None):
+        """Build a ret2csu ROPchain
+
+        Arguments:
+            edi, rsi, rdx: Three primary registers to populate
```

Comment:
We should probably check that the `edi` value can be packed with `p32()` (i.e., that it's a 32-bit value).